src/bundle/Resources/public/js/scripts/embedded.item.actions.js

```diff
@@ -0,0 +1,308 @@
+        const editItem = generateEditActionItem({ contentId, locationId, productCode, languages });
+
+        return {
+            groups: [
+                {
+                    id: 'default',
+                    items: [goToItem, editItem],
+                },
+            ],
+        };
+    };
+    const getLanguagesData = async ({ contentId, initialFunc = () => {}, callbackFunc = () => {} }) => {
+        try {
+            initialFunc();
+
+            const url = window.Routing.generate('ibexa.permission.limitation.language.content_edit', { contentInfoId: contentId });
+            const request = new Request(url, {
+                method: 'GET',
+                headers: { 'X-CSRF-Token': token },
+                mode: 'same-origin',
+                credentials: 'same-origin',
+            });
+            const response = await fetch(request);
+            const data = await ibexa.helpers.request.getJsonFromResponse(response);
+
+            callbackFunc();
+
+            return data.filter((language) => language.hasAccess);
+        } catch (error) {
+            ibexa.helpers.notification.showErrorNotification(error);
+        }
+    };
+    const getMenuData = ({ container, event }) => {
+        const { contentId, locationId, productCode, languageCodes } = container ? container.dataset : event.detail;
+        const parsedLanguageCodes = typeof languageCodes === 'string' ? JSON.parse(languageCodes) : languageCodes;
+        const languages = parsedLanguageCodes
+            ? parsedLanguageCodes.map((languageCode) => ({
+                  languageCode,
+                  name: adminUiLanguages[languageCode].name,
+              }))
+            : [];
+
+        return {
+            contentId: parseInt(contentId, 10),
+            locationId: parseInt(locationId, 10),
+            productCode,
+            languages,
+        };
+    };
+    const createMenu = async ({ triggerElement, container, contentId, locationId, productCode, languages }) => {
+        triggerElement.dataset.isMenuAttached = 1;
+
+        const mainContainer = container.closest('.ibexa-embedded-item-actions');
+        const menuLoader = mainContainer.querySelector('.ibexa-embedded-item-actions__loader');
+        const askForLanguagesData = Object.keys(languages).length !== 1;
+        const languagesData = askForLanguagesData
+            ? await getLanguagesData({
+                  contentId,
+                  initialFunc: showLoader.bind(null, { triggerElement, menuLoader }),
+                  callbackFunc: hideLoader.bind(null, { menuLoader }),
+              })
+            : languages;
+        const menuItems = generateMenuTreeItems({ contentId, locationId, productCode, languages: languagesData });
+        const menuInstance = new ibexa.core.MultilevelPopupMenu({
+            container,
+            triggerElement,
+        });
+        menuInstance.init();
```

Comment:
personally I would add here extra empty line before as it looks crunched, but your call
(I guess you wanted to group whole menuInstance logic, so that's only small suggestion)